cuda_next/CMakeLists.txt

```
@@ -0,0 +1,19 @@
+if (NOT DEFINED CCCL_ENABLE_CUDA_NEXT OR CCCL_ENABLE_CUDA_NEXT)
```

That seems strange, we should not build that by default but only if requested
```suggestion
if (CCCL_ENABLE_CUDA_NEXT)
```

---

I wanted tests compilation to work when running Cmake in CUDA_Next directory directly. Added CUDA_NEXT_TOPLEVEL_PROJECT similar to how other libraries do it instead.